src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -1228,6 +1221,11 @@ protected Engine newEngine(boolean skipTranslogRecovery, EngineConfig config) {
         return engineFactory.newReadWriteEngine(config, skipTranslogRecovery);
     }
 
+    /* create a new translog if needed. can return a null if not needed */
+    protected FsTranslog newTranslog(ShardId shardId, Settings indexSettings, IndexSettingsService indexSettingsService,
```

Comment:
Also, please annotate with `@IndexSettings Settings indexSettings`, there's nothing worse than `indexSettings` being renamed to `settings` at a later time and then not knowing which Settings it actually is.